tools/breaking-change-detector/rules/rules_field.go

```diff
@@ -14,29 +14,173 @@ type FieldRule struct {
+		oldType := getValueType(old.Type) + "." + getValueType(oldCasted.Type)
+		newType := getValueType(new.Type) + "." + getValueType(newCasted.Type)
+		message = strings.ReplaceAll(message, "{{oldType}}", oldType)
+		message = strings.ReplaceAll(message, "{{newType}}", newType)
+		return populateMessageContext(message, mc)
+	}
+
+	return ""
 }
 
 var fieldRule_BecomingRequired = FieldRule{
 	name:        "Field becoming Required Field",
-	definition:  "A field cannot become required as existing terraform modules may not have this field defined. Thus breaking their modules in sequential plan or applies.",
+	definition:  "A field cannot become required as existing terraform modules may not have this field defined. Thus breaking their modules in sequential plan or applies. If you are adding required to a field so a block won't remain empty this can cause two issues. First if it's a singular nested field the block may gain more fields later and it's not clear whether the field is actually required so it may be misinterpreted by future contributors. Second if users are defining empty blocks in existing configurations this change will break them. Consider these points in admittance of this type of change.",
 	message:     "Field {{field}} changed from optional to required on {{resource}}",
 	identifier:  "field-optional-to-required",
 	isRuleBreak: fieldRule_BecomingRequired_func,
 }
 
-func fieldRule_BecomingRequired_func(old, new *schema.Schema) bool {
+func fieldRule_BecomingRequired_func(old, new *schema.Schema, mc MessageContext) string {
+	message := mc.message
 	if !old.Required && new.Required {
-		return true
+		return populateMessageContext(message, mc)
 	}
 
-	return false
+	return ""
+}
+
+var fieldRule_BecomingComputedOnly = FieldRule{
+	name:        "Becoming a Computed only Field",
+	definition:  "While a field can go from Optional to Optional+Computed it cannot go from Required or Optional to only Computed. This transition would effectively make the field read-only thus breaking modules in sequential plan or applies where this field is defined in a configuration.",
+	message:     "Field {{field}} became Computed only on {{resource}}",
+	identifier:  "field-becoming-computed",
+	isRuleBreak: fieldRule_BecomingComputedOnly_func,
+}
+
+func fieldRule_BecomingComputedOnly_func(old, new *schema.Schema, mc MessageContext) string {
+	message := mc.message
+	// if the field is computed only already
+	// this rule doesn't apply
+	if old.Computed && !old.Optional {
+		return ""
+	}
+
+	if new.Computed && !new.Optional {
+		return populateMessageContext(message, mc)
+	}
+	return ""
+}
+
+var fieldRule_OptionalComputedToOptional = FieldRule{
+	name:        "Optional and Computed to Optional",
+	definition:  "A field cannot go from computed + optional to optional. On a sequential applies the terraform state will have the previously computed value. The value won't be present in the config, thus ultimately causing a diff.",
```

Comment:
```suggestion
	definition:  "A field cannot go from Computed + Optional to Optional. On a sequential "apply" the terraform state will have the previously computed value. The value won't be present in the config, thus ultimately causing a diff.",
```
what do you think of putting quotes around commands like `apply`? Unless backticks will show up correctly, i'm not sure. Either way, I'm pretty sure "applies" will be confusing vs "apply commands"

Reply:
I think its a good change !